### PR TITLE
Fix typos in API documentation

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to fix typos in the API documentation.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/plot.py [hunk 1]
```diff
@@ -206,8 +206,9 @@ def plot(self, data=None, x=None, y=None, size=None, direction=None, **kwargs):
     {label}
     {perspective}
     {transparency}
-        *transparency* can also be a 1d array to set varying transparency
-        for symbols, but this option is only valid if using x/y.
+        ``transparency`` can also be a 1d array to set varying
+        transparency for symbols, but this option is only valid if using
```
Comment (seisman) [inline pygmt/src/plot.py:210]:
```suggestion
        transparency for symbols, but this parameter is only valid if using
```

File: pygmt/src/plot3d.py [hunk 2]
```diff
@@ -176,8 +176,9 @@ def plot3d(
     {label}
     {perspective}
     {transparency}
-        *transparency* can also be a 1d array to set varying transparency
-        for symbols, but this option is only valid if using x/y/z.
+        ``transparency`` can also be a 1d array to set varying
+        transparency for symbols, but this option is only valid if using
```
Comment (seisman) [inline pygmt/src/plot3d.py:180]:
```suggestion
        transparency for symbols, but this parameter is only valid if using
```

File: pygmt/src/text.py [hunk 3]
```diff
@@ -159,8 +160,9 @@ def text_(
     {incols}
     {perspective}
     {transparency}
-        *transparency* can also be a 1d array to set varying transparency
-        for texts, but this option is only valid if using x/y/text.
+        ``transparency`` can also be a 1d array to set varying
+        transparency for texts, but this option is only valid if using
```
Comment (seisman) [inline pygmt/src/text.py:164]:
```suggestion
        transparency for texts, but this parameter is only valid if using
```

File: pygmt/src/text.py [hunk 4]
```diff
@@ -122,13 +122,14 @@ def text_(
...
         [Default is 15% of the font size]. Only used if ``pen`` or ``fill`` are
         specified. Append the unit you want (*c* for cm, *i* for inch, or *p*
-        for point; if not given we consult **PROJ_LENGTH_UNIT**) or *%* for a
-        percentage of the font size. Optionally, use modifier **+t** to set
-        the shape of the textbox when using ``fill`` and/or ``pen``. Append
-        lower case **o** to get a straight rectangle [Default is **o**]. Append
-        upper case **O** to get a rounded rectangle. In paragraph mode
-        (*paragraph*) you can also append lower case **c** to get a concave
-        rectangle or append upper case **C** to get a convex rectangle.
+        for point; if not given we consult :gmt_term:`PROJ_LENGTH_UNIT`)
```
Comment (seisman) [inline pygmt/src/text.py:125]:
```suggestion
        for point; if not given we consult :gmt-term:`PROJ_LENGTH_UNIT`)
```

File: pygmt/src/plot.py (hunk 1 above)
Comment (yvonnefroehlich) [reply pygmt/src/plot.py:211]: I thought `option` refers to the possibility to use varying transparency, not using the `transparency` parameter in general.

**Review by seisman — APPROVED** [review]

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**